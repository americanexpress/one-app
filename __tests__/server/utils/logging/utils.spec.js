/*
 * Copyright 2019 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  printStatusCode,
  printStatusMessage,
  printDurationTime,
} from '../../../../src/server/utils/logging/utils';

jest.mock('colorette');

describe('utils', () => {
  describe('printStatusCode', () => {
    it('prints a 200 status code as green', () => {
      expect(printStatusCode({ request: { statusCode: 200 } })).toMatchSnapshot();
    });

    it('prints a 300 status code as blue', () => {
      expect(printStatusCode({ request: { statusCode: 300 } })).toMatchSnapshot();
    });

    it('prints a 400 status code as yellow', () => {
      expect(printStatusCode({ request: { statusCode: 400 } })).toMatchSnapshot();
    });

    it('prints a 500 status code as red', () => {
      expect(printStatusCode({ request: { statusCode: 500 } })).toMatchSnapshot();
    });

    it('prints a null status code as black text on red background', () => {
      expect(printStatusCode({ request: { statusCode: null } })).toMatchSnapshot();
    });
  });

  describe('printStatusMessage', () => {
    it('returns the status text when present', () => {
      expect(printStatusMessage({ request: { statusCode: null, statusText: 'OK' } })).toBe('OK');
    });

    it('returns status text based off the status code when present', () => {
      expect(printStatusMessage({ request: { statusCode: 200, statusText: null } })).toBe('OK');
    });

    it('returns a black on red background timed out message when the text and code do not map to text', () => {
      expect(
        printStatusMessage({ request: { statusCode: null, statusText: null } })
      ).toMatchSnapshot();
    });
  });

  describe('printDurationTime', () => {
    it('prints good times in green', () => {
      expect(printDurationTime({ request: { timings: { duration: 99 } } })).toMatchSnapshot();
    });

    it('prints acceptable times in yellow', () => {
      expect(printDurationTime({ request: { timings: { duration: 999 } } })).toMatchSnapshot();
    });

    it('prints long times in red', () => {
      expect(printDurationTime({ request: { timings: { duration: 5999 } } })).toMatchSnapshot();
    });

    it('prints unacceptable times in black on a red background', () => {
      expect(printDurationTime({ request: { timings: { duration: 6000 } } })).toMatchSnapshot();
    });
  });
});
