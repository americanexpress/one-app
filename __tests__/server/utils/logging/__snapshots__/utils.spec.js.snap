// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils coloredLevels has a blue log 1`] = `"<blue>log</blue>"`;

exports[`utils coloredLevels has a gray info 1`] = `"<gray>info</gray>"`;

exports[`utils coloredLevels has a red error 1`] = `"<red>error</red>"`;

exports[`utils coloredLevels has a yellow warn 1`] = `"<yellow>warn</yellow>"`;

exports[`utils printDurationTime prints acceptable times in yellow 1`] = `"<yellow>999</yellow>"`;

exports[`utils printDurationTime prints good times in green 1`] = `"<green>99</green>"`;

exports[`utils printDurationTime prints long times in red 1`] = `"<red>5999</red>"`;

exports[`utils printDurationTime prints unacceptable times in black on a red background 1`] = `"<black><bgRed>6000</bgRed></black>"`;

exports[`utils printStatusCode prints a 200 status code as green 1`] = `"<green>200</green>"`;

exports[`utils printStatusCode prints a 300 status code as blue 1`] = `"<blue>300</blue>"`;

exports[`utils printStatusCode prints a 400 status code as yellow 1`] = `"<yellow>400</yellow>"`;

exports[`utils printStatusCode prints a 500 status code as red 1`] = `"<red>500</red>"`;

exports[`utils printStatusCode prints a null status code as black text on red background 1`] = `"<black><bgRed>null</bgRed></black>"`;

exports[`utils printStatusMessage returns a black on red background timed out message when the text and code do not map to text 1`] = `"<black><bgRed>Timed Out?</bgRed></black>"`;
